lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -2,7 +2,7 @@ import DocumentModel from './models/DocumentModel';
 import DidState from '../../models/DidState';
```

Comment:
I am skipping this entire file for now as it is pending a bunch of code revert requested.

Reply:
Transmute, SecureKey and everyone using Linked Data Proofs and Verifiable Presentations rely on verification relationships beyond `authentication`... see:

https://w3c.github.io/did-core/#verification-relationships

We have support them in our fork here:

https://github.com/transmute-industries/sidetree.js/blob/master/packages/common/src/enums/PublicKeyPurpose.ts#L23

You can see the go implementation supports them here:

https://github.com/trustbloc/sidetree-core-go/blob/28fabda71792f0bf59735ddbc9fa38bebcca1a32/pkg/dochandler/transformer/didtransformer/transformer.go#L130

Yet the sidetree reference implementation and spec do not align with did core on this front.

The Document compose change set most applies to public_keys vs `publicKey` and `service_endpoints` vs `service`... I did not implement support for new verification relationships in this code base, but I did correct the enum and cover the cases documented in did core today, because.... rather fix them all at once while making other breaking changes.

Reply:
I disagree. But since this is contentious and technically unrelated to the casing change, please leave this part of the change out for a separate discussion so we can at least push the agreed upon change as at least a small win.

Reply:
Seems like the consensus on the call was to take these changes.

Reply:
It's my preference that we use the terms defined in the did core specification, but also adhere to the plurality rule that's mentioned in all of those style guides. 

This would leave us with 
publicKeys, 
verificationRelationships
services

The reason I'm preferring this is because the names of the properties will be easy to map as a developer that they're the similar to what's in did-core not leaving it up to interpretation and getting very close to the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment), but we place the style guide above all as the preferred way to implement the code without exception. This will go a long way to resolving conflicts for us because then we can always have the style guide as an independent arbitrator of truth. 

On a side note, the problem is that the did-core specification isn't using plural nouns for arrays, but given that ship has sailed, let's at least make sure we get it right here.